### ISSUE TITLE
Add GAN Mask R-CNN model to detectron2 model zoo

### ISSUE THREAD

**Issue by armenforget** [facebookresearch/detectron2#3829]
## 🚀 Feature
Add GAN Mask R-CNN model to detectron2 model zoo

## Motivation & Examples

Tell us why the feature is useful.
Some research has been done that demonstrates GAN Mask R-CNN is able to produce crisper edges compared to the non-GAN version.
- See [Instance Semantic Segmentation Benefits from Generative Adversarial Networks](https://arxiv.org/abs/2010.13757) for research paper.
- See [GAN_Mask-RCNN](https://github.com/quangle2110/GAN_Mask-RCNN) for code from research paper.

Describe what the feature would look like, if it is implemented.
```python
model_file = "COCO-InstanceSegmentation/gan_mask_rcnn_R_50_FPN.yaml"
cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file(model_file))

# omitting dataset registration and other cfg parameters
trainer = DefaultTrainer(cfg)
trainer.resume_or_load(resume=False)
trainer.train()
```

## Note

We only consider adding new features if they are relevant to many users.

If you request implementation of research papers -- we only consider papers that have enough significance and prevalance in the object detection field.

We do not take requests for most projects in the `projects/` directory, because they are research code release that is mainly for other researchers to reproduce results.

"Make X faster/accurate" is not a valid feature request. "Implement a concrete feature that can make X faster/accurate" can be a valid feature request.

Instead of adding features inside detectron2,
you can implement many features by [extending detectron2](https://detectron2.readthedocs.io/tutorials/extend.html).
The [projects/](https://github.com/facebookresearch/detectron2/tree/main/projects/) directory contains many of such examples.
